### ISSUE TITLE
UnusedImportsCheck does not detect parameter types in javadoc block tags

### ISSUE THREAD

**Issue by silentj** [checkstyle/checkstyle#4606]
```
/var/tmp $ javac InputUnusedImportWithBlockMethodParameters.java 
[no output]

/var/tmp $ cat InputUnusedImportWithBlockMethodParameters.java 
package com.puppycrawl.tools.checkstyle.checks.imports.unusedimports;

import java.util.Collection;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.TimeUnit;

public class InputUnusedImportWithBlockMethodParameters {

/**
 * @see ExecutorService#invokeAll(Collection, long, TimeUnit)
 */
    public int calculate() {
        return 0;
    }
}

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="UnusedImports">
      <property name="processJavadoc" value="true"/>
    </module>
  </module>
</module>

/var/tmp $ java -jar checkstyle-8.0-all.jar -c config.xml InputUnusedImportWithBlockMethodParameters.java 
Starting audit...
[ERROR] /private/var/tmp/InputUnusedImportWithBlockMethodParameters.java:3:8: Unused import - java.util.Collection. [UnusedImports]
[ERROR] /private/var/tmp/InputUnusedImportWithBlockMethodParameters.java:5:8: Unused import - java.util.concurrent.TimeUnit. [UnusedImports]
Audit done.
Checkstyle ends with 2 errors.
```
---------------

This should not be an error.

`UnusedImportCheck` fails to find `Collection` and `TimeUnit` in the javadoc. If these classes are unreferenced by the code, the imports are reported as unused.

This style works fine for inline javadoc tags (`{@link ExecutorService#invokeAll(Collection, long, TimeUnit)}`) but the difference in how [block](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java#L259) and [inline](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java#L255) tags are parsed means method parameter types aren't found for block tags.

**Comment by silentj** [comment]
Let me know if there's anything else I need to do to get this issue approved. Thanks.

**Comment by romani** [comment]
@silentj ,

it is always good to read issue rules - https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE.md

Issue should properly formatted, in other case it will stay unapproved till other more simple to reproduce issues are resolved, one day in future. 
http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by silentj** [comment]
Sorry, @romani  - Issue description updated to conform to the guidelines.

**Comment by romani** [comment]
type detection is not supported by Checkstyle, we can not recognize types, we could match them only by string values, so all approaches will be error-prone. But at least we will generate less false-positives that annoy users more than false-negatives.

**Comment by romani** [comment]
@silentj , thanks a lot for your updates, it is pleasure read your detailed PRs and good testing.
